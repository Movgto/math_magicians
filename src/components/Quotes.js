import PropTypes from 'prop-types';
import '../stylesheets/quotes.scss';
import { useState, useEffect } from 'react';

export default function Quotes({ apiKey }) {
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      setLoading(true);

      try {
        const response = await fetch('https://api.api-ninjas.com/v1/quotes', {
          method: 'GET',
          headers: { 'X-Api-Key': apiKey },
          contentType: 'application/json',
        });

        const json = await response.json();
        setData(json);
      } catch (error) {
        setError(true);
      }

      setLoading(false);
    };

    fetchData();
  }, [apiKey]);

  if (error) {
    return (
      <div id="quotes" data-testid="quotes">
        <h3 className="error">
          Couldn&apost get data from source:
        </h3>
      </div>
    );
  }

  if (loading) {
    return (
      <div id="quotes" data-testid="quotes">
        <h3>Loading...</h3>
      </div>
    );
  }

  if (data.length > 0) {
    return (
      <div id="quotes" data-testid="quotes">
        <p>{data[0].quote}</p>
        <h3>
          {'- '}
          {data[0].author}
        </h3>
      </div>
    );
  }
}

Quotes.propTypes = {
  apiKey: PropTypes.string.isRequired,
};
