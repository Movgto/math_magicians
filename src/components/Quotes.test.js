import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quotes from './Quotes';

describe('Snapshot', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(<Quotes />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test quotes component', () => {
  it('loading state', async () => {
    const { findByTestId } = render(<Quotes apiKey="123" />);
    const quotes = await findByTestId('quotes');
    expect(quotes.querySelector('h3').textContent).toBe('Loading...');
  });
});
