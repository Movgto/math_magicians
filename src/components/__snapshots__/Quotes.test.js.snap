// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders properly 1`] = `null`;
